salt/elife-xpub/scripts/setup-database.sh

```diff
@@ -1,19 +1,19 @@
 #!/bin/bash
 set -e
 
-DC_COMMAND="docker-compose -f docker-compose.yml -f docker-compose.formula.yml"
-DB_CREATED_COMMAND="psql -c \"SELECT 'public.entities'::regclass\""
-DB_ENV="-e PGHOST=${PGHOST} -e PGPORT=${PGPORT} -e PGUSER=${PGUSER} -e PGDATABASE=${PGDATABASE} -e PGPASSWORD=${PGPASSWORD}"
-SETUP_ARGS="--username={{ pillar.elife_xpub.database.user }} --password={{ pillar.elife_xpub.database.password }} --email={{ pillar.elife_xpub.database.email }}"
+dc="docker-compose -f docker-compose.yml -f docker-compose.formula.yml"
+db_env="-e PGHOST=${PGHOST} -e PGPORT=${PGPORT} -e PGUSER=${PGUSER} -e PGDATABASE=${PGDATABASE} -e PGPASSWORD=${PGPASSWORD}"
+recreate="dropdb ${PGDATABASE} && createdb ${PGDATABASE}"
+
+# Removing setting up a user (via clobber) in PubSweet as at the moment this is no longer necessary
+# This may be re-instated after the work to upgrade to the latest PubSweet version.
+# Ref: elifesciences/elife-xpub/issues/1920
+#
+# SETUP_ARGS="--username={{ pillar.elife_xpub.database.user }} --password={{ pillar.elife_xpub.database.password }} --email={{ pillar.elife_xpub.database.email }}"
 
 if [ ! -z "${DROP}" ]; then
-    ${DC_COMMAND} run --rm app /bin/bash -c "npx pubsweet setupdb ${SETUP_ARGS} --clobber"
-    exit
+    ${dc} run --rm ${db_env} postgres /bin/bash -c "${recreate}"
```

Comment:
Does not seem equivalent to `clobber`, which [deletes the tables rather than the database](https://gitlab.coko.foundation/pubsweet/pubsweet/blob/master/packages/db-manager/src/commands/create-tables.js#L17). Especially on RDS, the database and the `public` schema within it are a mess of ownership between AWS's superuser and the almost-superuser `root`, so should in general not be touched by an application formula.

Reply:
Correctly doesn't exit because it will need the `migrate` later.

Reply:
Just updating and contemplating using the pubsweet cleaning function ... but that appears to drop the whole public schema:
```
const dbCleaner = async options => {
  await db.raw('DROP SCHEMA public CASCADE;')
  await db.raw('CREATE SCHEMA public;')
  await db.raw('GRANT ALL ON SCHEMA public TO public;')
  await migrate(options)
  logger.info('Dropped all tables and ran all migrations')
  return true
}

``` 
:thinking: ... what would you suuggest?

Reply:
The implementation I linked is a bit complex but it correctly `DROP`s the tables - though iterating over tables in pure SQL would be a nightmare.

Reply:
At least on RDS, the schema owner is `root` so it should be possible to drop that (which includes all tables) and recreate it:
```
elifexpubstaging=> select schema_owner from information_schema.schemata where schema_name='public';
 schema_owner 
--------------
 root
```
Have to cross-check with what happens in a `postgres` container.

Reply:
In that case:
```
root@e3e5d8d0f9ed:/# PGUSER=xpub psql
psql (10.7 (Debian 10.7-1.pgdg90+1))
Type "help" for help.
xpub=# select schema_owner from information_schema.schemata where schema_name='public';
 schema_owner 
--------------
 xpub
(1 row)
```
The user `xpub` is the one the application uses to connect to the `postgres` container, and is the owner of the schema. So the simplest solution here is to drop and recreate the `public` schema and it should all be fine with the right ownership.